docs/developer/extending/apps/developming-with-tunnels.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: Developing apps with tunnels
+---
+
+Saleor apps can be developed fully locally or using Saleor deployed to the cloud (either Saleor Cloud or a self-hosted instance on the internet).
+
+This article describes developing apps using tunnelling.
+
+## How does it work
+
+Saleor must communicate with your app, but it can't reach your localhost. Tunnelling software creates a proxy,
+allowing access to a specific port from localhost to a publically accessible URL
+
+<!-- Consider diagram -->
```

Comment:
```mermaid
flowchart TB
    a[Dev server]
    t[Tunnel]
    c[Saleor cloud]
    
    c-->t
    a --> t
    a-->c
    c-- Can't reach dev env --->a
    t-->c
    t --> a
```

Heres one, but don't love it. Ideas to improve?

Reply:
Maybe something like in this article 
https://www.atwix.com/development-tools/ngrok/

Reply:
```mermaid
flowchart TB
    a[Dev server]
    t[Tunnel]
    c[Saleor cloud]
    
    c-->t
    a<-.-> t
    t-->c
```

Or

```mermaid
flowchart LR
    a[Dev server]
    t[Tunnel]
    c[Saleor cloud]
    e[External services]
    
    subgraph Local environment
        a
    end

    subgraph Internet
        t
        c
        e
    end

    a<-.-> t
    t<-->c
    t<-->e
```

Reply:
I like both :) @peelar ?